flipflOS/Kernel.cs

```diff
@@ -275,6 +285,79 @@ public ushort readFromMemory(String[] args)
 
             return mem.readAt(index);
         }
+        public void removeFile(String path)
+        {
+            Directory startingdir = currentdir; //speichert startverzeichnis um später zurückzukommen
+            String[] seperatedbyslash = path.Split("/"); //teilt pfad in unterOrdner
+            String file = seperatedbyslash[seperatedbyslash.Length - 1]; //speichert Dateiname
+            for(int i = 0;i < seperatedbyslash.Length - 1; i++)
+            {
+                changeDir(seperatedbyslash[i]); //geht zum Pfad wo die Datei ist
+            }
+            currentdir.removeFile(file); //löscht File
+            currentdir = startingdir; //geht wieder zum Startverzeichnis
+        }
+        public void moveFile(String[] args)
+        {
+            String path = args[1];        //nimmt die argumente
+            String destination = args[2];
+
+            Directory startingdir = currentdir;
+            String[] seperatedbyslash = path.Split("/"); //teilt den ersten path mit den Slashs
+            String file = seperatedbyslash[seperatedbyslash.Length - 1];
+
+            for (int i = 0; i < seperatedbyslash.Length - 1; i++)
+            {
+                changeDir(seperatedbyslash[i]); //bewegt currentdir zur path von der Datei
+            }
+            if(currentdir.getFile(file) == null)
+            {
+                return;
+            }
+            Directory.File movingFile = currentdir.getFile(file); //speichert Datei in einer temporären Variable
+            Directory firstFileDirectory = currentdir; //speichert directory von File ab, falls Zieldir bereits Datei mit namen beinhaltet.
+            currentdir = startingdir; //fängt von vorne an
+
+            seperatedbyslash = destination.Split("/"); //teilt zielPfad 
+            for (int i = 0; i < seperatedbyslash.Length; i++)
+            {
+                changeDir(seperatedbyslash[i]); //bewegt currentdir zum Zielpfad
+            }
+            if (currentdir.addFile(movingFile))  //speichert Datei dort, falls es funtktioniert dann
+            {
+                firstFileDirectory.removeFile(file); //wird die Datei aus dem vorherigem Verzeichnis gelöscht
+            }
+            currentdir = startingdir; //geht wieder zum Startverzeichnis
+        }
+        public void copyFile(String[] args)
+        {
+            String path = args[1];        //nimmt die argumente
+            String destination = args[2];
+
+            Directory startingdir = currentdir;
+            String[] seperatedbyslash = path.Split("/"); //teilt den ersten path mit den Slashs
+            String file = seperatedbyslash[seperatedbyslash.Length - 1];
+
+            for (int i = 0; i < seperatedbyslash.Length - 1; i++)
+            {
+                changeDir(seperatedbyslash[i]); //bewegt currentdir zur path von der Datei
+            }
+            if (currentdir.getFile(file) == null)
+            {
+                return;
```

Comment:
Vielleicht nen output, dass der file nicht existiert

Reply:
gibt es schon. der Output wird in der getFile funktion geprinted.